File: src/main/java/com/brasilburger/model/CommandeItem.java
package com.brasilburger.model;

import com.brasilburger.model.enums.TypeItem;

/**
 * Classe représentant un article dans une commande
 */
public class CommandeItem {
    private int id;
    private int commandeId;
    private TypeItem typeItem;
    private int itemId;
    private int quantite;
    private double prixUnitaire;
    private double sousTotal;

    public CommandeItem() {
    }

    public CommandeItem(int commandeId, TypeItem typeItem, int itemId, int quantite, double prixUnitaire) {
        this.commandeId = commandeId;
        this.typeItem = typeItem;
        this.itemId = itemId;
        this.quantite = quantite;
        this.prixUnitaire = prixUnitaire;
        this.sousTotal = quantite * prixUnitaire;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getCommandeId() { return commandeId; }
    public void setCommandeId(int commandeId) { this.commandeId = commandeId; }

    public TypeItem getTypeItem() { return typeItem; }
    public void setTypeItem(TypeItem typeItem) { this.typeItem = typeItem; }

    public int getItemId() { return itemId; }
    public void setItemId(int itemId) { this.itemId = itemId; }

    public int getQuantite() { return quantite; }
    public void setQuantite(int quantite) { this.quantite = quantite; }

    public double getPrixUnitaire() { return prixUnitaire; }
    public void setPrixUnitaire(double prixUnitaire) { this.prixUnitaire = prixUnitaire; }

    public double getSousTotal() { return sousTotal; }
    public void setSousTotal(double sousTotal) { this.sousTotal = sousTotal; }

    @Override
    public String toString() {
        return "CommandeItem{" +
                "id=" + id +
                ", typeItem=" + typeItem +
                ", quantite=" + quantite +
                ", sousTotal=" + sousTotal +
                '}';
    }
}

File: src/Entity/CommandeItem.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'commande_items')]
class CommandeItem
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Commande::class, inversedBy: 'items')]
    #[ORM\JoinColumn(name: 'commande_id', referencedColumnName: 'id', nullable: false, onDelete: 'CASCADE')]
    private ?Commande $commande = null;

    #[ORM\Column(length: 50)]
    private string $typeItem;

    #[ORM\Column(type: 'integer')]
    private int $itemId;

    #[ORM\Column(type: 'integer')]
    private int $quantite;

    #[ORM\Column(name: 'prix_unitaire', type: 'decimal', precision: 10, scale: 2, nullable: true)]
    private ?string $prixUnitaire = null;

    #[ORM\Column(name: 'sous_total', type: 'decimal', precision: 10, scale: 2, nullable: true)]
    private ?string $sousTotal = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(?Commande $commande): self
    {
        $this->commande = $commande;

        return $this;
    }

    public function getTypeItem(): string
    {
        return $this->typeItem;
    }

    public function setTypeItem(string $typeItem): self
    {
        $this->typeItem = $typeItem;

        return $this;
    }

    public function getItemId(): int
    {
        return $this->itemId;
    }

    public function setItemId(int $itemId): self
    {
        $this->itemId = $itemId;

        return $this;
    }

    public function getQuantite(): int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;

        return $this;
    }

    public function getPrixUnitaire(): ?string
    {
        return $this->prixUnitaire;
    }

    public function setPrixUnitaire(?string $prixUnitaire): self
    {
        $this->prixUnitaire = $prixUnitaire;

        return $this;
    }

    public function getSousTotal(): ?string
    {
        return $this->sousTotal;
    }

    public function setSousTotal(?string $sousTotal): self
    {
        $this->sousTotal = $sousTotal;

        return $this;
    }
}
